src_base/xeve_args.c

```diff
@@ -0,0 +1,2313 @@
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_16, param->rpl1[16]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_17, param->rpl1[17]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_18, param->rpl1[18]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_19, param->rpl1[19]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_20, param->rpl1[20]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_21, param->rpl1[21]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_22, param->rpl1[22]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_23, param->rpl1[23]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_24, param->rpl1[24]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_RPL1_25, param->rpl1[25]);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_INTER_SLICE_TYPE, &param->inter_slice_type);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_PIC_CRARG_FLAG, &param->picture_cropping_flag);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_PIC_CRARG_LEFT, &param->picture_crop_left_offset);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_PIC_CRARG_RIGHT, &param->picture_crop_right_offset);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_PIC_CRARG_TOP, &param->picture_crop_top_offset);
+    ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, ARG_PIC_CRARG_BOTTOM, &param->picture_crop_bottom_offset);
+#endif
+    args->initialized = 1;
+
+    return XEVE_OK;
+}
+
+int xeve_args_parse(ARGS_PARSER * args, int argc, const char* argv[], char ** errstr)
+{
+    int i, ret = 0, idx = 0;
+    const char* fname_cfg = NULL;
+    FILE* fp;
+
+    int num_configs = 0;
+    int pos_conf_files[ARGS_MAX_NUM_CONF_FILES];
+
+     // do some checks
+    if(!args || argc<=0 || !argv)
+        return XEVE_ERR_INVALID_ARGUMENT;
+
+    if( !args->initialized )
+        return XEVE_ERR;
+
+    memset(&pos_conf_files, -1, sizeof(int) * ARGS_MAX_NUM_CONF_FILES);
+
+    /* config file parsing */
+    for (i = 1; i < argc; i++)
+    {
+        if (!strcmp(argv[i], "--"ARGS_KEY_LONG_CONFIG))
+        {
+            if (i + 1 < argc)
+            {
+                num_configs++;
+                pos_conf_files[num_configs - 1] = i + 1;
+            }
+        }
+    }
+    for (int i = 0; i < num_configs; i++)
+    {
+        fname_cfg = argv[pos_conf_files[i]];
+        if (fname_cfg)
+        {
+            fp = fopen(fname_cfg, "r");
+            if (fp == NULL) return -1; /* config file error */
+
+            if (args_parse_cfg(fp, args->opts, 1))
+            {
+                fclose(fp);
+                return -1; /* config file error */
+            }
+            fclose(fp);
+        }
+    }
+    /* command line parsing */
+    while (1)
+    {
+        ret = args_parse_cmd(argc, argv, args->opts, &idx, errstr);
+        if (ret <= 0) break;
+    }
+    return ret;
+}
+
+int xeve_args_check_mandatory(ARGS_PARSER * args, char ** err_arg)
+{
+    ARGS_OPT * o = args->opts;
+
+    // do some checks
+    if(!args)
```

Comment:
This check should be before the line ARGS_OPT * o = args->opts;

Reply:
DONE